superset-frontend/src/explore/components/controls/DndColumnSelectControl/DndColumnSelect.tsx

```diff
@@ -127,14 +127,7 @@ export function DndColumnSelect(props: DndColumnSelectProps) {
   );
 
   const popoverOptions = useMemo(
-    () =>
-      Object.values(options).filter(
-        col =>
-          !optionSelector.values
-            .filter(isColumnMeta)
-            .map((val: ColumnMeta) => val.column_name)
-            .includes(col.column_name),
```

Comment:
Did you confirm that we don't have other side effects by removing this filter?

If we really don't need this filter, can you add the reason into PR description so others can get an idea?

Reply:
It was supposed to filter out the current value from options when editing metric/column/filter. I suppose we don't really need it so lgtm 👍